python/rapidsmpf/rapidsmpf/integrations/single.py

```diff
@@ -0,0 +1,222 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+"""Shuffler integration for single-worker pylibcudf execution."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, cast
+
+from rapidsmpf.communicator.single import new_communicator
+from rapidsmpf.config import Options
+from rapidsmpf.integrations.core import (
+    WorkerContext,
+    extract_partition,
+    get_new_shuffle_id,
+    get_shuffler,
+    insert_partition,
+    rmpf_worker_setup,
+)
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
+    from rapidsmpf.integrations.core import ShufflerIntegration
+
+
+# Local single-worker context
+class _SingleWorker:
+    """Mutable single-worker utility class."""
+
+    context: WorkerContext | None = None
+
+
+_single_rapidsmpf_worker: _SingleWorker = _SingleWorker()
+
+
+def _get_single_worker_context(worker: _SingleWorker | None = None) -> WorkerContext:
```

Comment:
*nit*: This function is private to this module, which is already called `single`, why do we need to repeat `Single` everywhere?

How about `_worker_context`?